File: statsig-dotnet/src/Statsig/ClientInitResponseOptions.cs
using System;
using System.Text;
using System.Text.Json;
using Newtonsoft.Json.Linq;
using Newtonsoft.Json;

namespace Statsig
{
    /// <summary>
    /// Configuration options for the GetClientInitializeResponse method in the Statsig Server SDK.
    /// </summary>
    public class ClientInitResponseOptions
    {
        [JsonProperty("hash_algorithm")]
        public string? HashAlgorithm { get; set; }
        [JsonProperty("client_sdk_key")]
        public string? ClientSDKKey { get; set; }
        [JsonProperty("include_local_overrides")]
        public bool IncludeLocalOverrides { get; set; }
        [JsonProperty("remove_id_type")]
        public bool RemoveIdType { get; set; }

        public ClientInitResponseOptions()
        {
        }
    }
}

File: statsig-go/client_init_response_options.go
package statsig_go_core

type ClientInitResponseOptions struct {
	HashAlgorithm           *string   `json:"hash_algorithm,omitempty"`
	ClientSDKKey            *string   `json:"client_sdk_key,omitempty"`
	IncludeLocalOverrides   *bool     `json:"include_local_overrides,omitempty"`
	ResponseFormat          *string   `json:"response_format,omitempty"`
	FeatureGateFilter       *[]string `json:"feature_gate_filter,omitempty"`
	ExperimentFilter        *[]string `json:"experiment_filter,omitempty"`
	DynamicConfigFilter     *[]string `json:"dynamic_config_filter,omitempty"`
	LayerFilter             *[]string `json:"layer_filter,omitempty"`
	ParamStoreFilter        *[]string `json:"param_store_filter,omitempty"`
	RemoveIdType            *bool     `json:"remove_id_type,omitempty"`
	RemoveDefaultValueGates *bool     `json:"remove_default_value_gates,omitempty"`
}
